### ISSUE TITLE
help to build it locally, build fails due to missing signing store

### ISSUE THREAD

**Issue by uwla** [synonymdev/bitkit-react-native#1521]
I'd like to ask for help to build bitkit from source.

I tried to build it but go some errors related to the signing store.

I have the following environment:

- Android SDK 33
- Java JDK 11.0.21
- node v20.1
- yarn v1.21

When running `yarn bundle` I get the error:

```
> Task :app:packageRelease FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:packageRelease'.
> A failure occurred while executing com.android.build.gradle.tasks.PackageAndroidArtifact$IncrementalSplitterRunnable
   > SigningConfig "release" is missing required property "storeFile".
```

I also tried:

```
npx react-native build-android --mode=release
```

And got:

```
> Task :app:signReleaseBundle FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:signReleaseBundle'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.FinalizeBundleTask$BundleToolRunnable
   > java.lang.NullPointerException (no error message)
```

I found out the problem lies in `android/app/build.gradle`, due to some properties are not loaded in the `if` block:

```
signingCOnfigs{
     ...
     release {
            if (project.hasProperty('BITKIT_UPLOAD_STORE_FILE')) {
                storeFile file(BITKIT_UPLOAD_STORE_FILE)
                storePassword BITKIT_UPLOAD_STORE_PASSWORD
                keyAlias BITKIT_UPLOAD_KEY_ALIAS
                keyPassword BITKIT_UPLOAD_KEY_PASSWORD
            }
        }
},
buildTypes {
        ...
        release {
            // Caution! In production, you need to generate your own keystore file.
            // see https://reactnative.dev/docs/signed-apk-android.
            signingConfig signingConfigs.release
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
        }
}
```

I tried to solve it by generating my own keystore, as pointed out by React docs:

```
cd bitkit/
keytool -genkey -v -keystore keyname.keystore -alias keyalias -keyalg RSA -keysize 2048
```

And answered `keypass` when prompted for the password.

Then I added the following vars to `.env` (which was copied from `.env.development.template`):

```
BITKIT_UPLOAD_STORE_FILE=keyname.keystore
BITKIT_UPLOAD_STORE_PASSWORD=keypass
BITKIT_UPLOAD_KEY_ALIAS=keyalias
BITKIT_UPLOAD_KEY_PASSWORD=keypass
```

After that, I tried building bitkit again but I got the same errors.

I also tried to copy the file `bitkit/keyname.keystore` to `bitkit/android/app` and did not work as well.

My last attempt was to export these `envvars` to the shell: `export BITKIT_UPLOAD_STORE_FILE=keyname.keystore` and the others too. This didn't work.

Can somebody help here please?

I can provide a "how to reproduce" if needed.

**Comment by limpbrains** [comment]
Hi

you can refer to [RN documentation](https://reactnative.dev/docs/signed-apk-android#setting-up-gradle-variables) on how to do it properly

When debugging I usually just create 
```
~/.gradle/gradle.properties
```
with
```
BITKIT_UPLOAD_STORE_FILE=debug.keystore
BITKIT_UPLOAD_STORE_PASSWORD=android
BITKIT_UPLOAD_KEY_ALIAS=androiddebugkey
BITKIT_UPLOAD_KEY_PASSWORD=android
```

**Comment by uwla** [comment]
Thank you! It worked!

I had missplaced the envvars in `.env` instead of `~/.gradle/gradle.properties`...

So, before I close this issue, would a PR, adding details about building bitkit from source to the documentation, be welcome? So that other devs won't face this issue in the future. I also noticed the `BUILD.md` talks about the command `yarn setup-android`, but the script `setup-android` does not exist on `package.json`, so the documentation is outdated.

If that sounds good, I can make the PR. Just let me know.
